spectral_cube/stokes_spectral_cube.py

```diff
@@ -170,6 +196,59 @@ def _new_cube_with(self, stokes_data=None,
 
         return cube
 
+    def transform_basis(self, stokes_basis=''):
+        """
+        The goal of this function is to transform the stokes basis of the cube to
+        the one specified if possible. In principle one needs at least two related
+        stokes of a given basis for the transformation to be possible. At this moment
+        we are limiting it to the cases where all four stokes parameters in a given
+        basis be available within the cube. 
+        """
+        if self.shape[0] < 4:
+            errmsg = "Transformation of a subset of Stokes axes is not yet supported."
+            raise NotImplementedError(errmsg)
+        elif stokes_basis == "Generic":
+            data =  self._stokes_data
+        elif self.stokes_type == "Linear":
+            if stokes_basis == "Circular":
+                errmsg = "Transformation from %s to %s is not yet supported."% ("Linear","Circular")
+                raise NotImplementedError(errmsg)
+            if stokes_basis == "Sky":
+                data = dict(I = SpectralCube(self._stokes_data['XX'] + self._stokes_data['YY'], wcs = self.wcs, mask=self._mask['XX']|self._mask['YY']),
```

Comment:
Instead of using `SpectralCube` here (and below), you should use `self.__class__`.

Reply:
Also, probably the mask combination should be `&` combination, not `|`

Reply:
It has to be `|` so that it is masked channel if it is masked in either of the polarizations right ?

Reply:
You are right on both accounts. Fixed it.